### ISSUE TITLE
Injected code can be rejected due to CSP with Firefox

### ISSUE THREAD

**Issue by vincentbernat** [philc/vimium#2926]
Hey!

When a site has a CSP policy for script-src not including "self", we get this error:

```
Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src https://report-uri.com https://api.stripe.com https://js.stripe.com https://www.google.com https://www.gstatic.com https://cdnjs.cloudflare.com https://ajax.cloudflare.com https://www.google-analytics.com”). Source: (function() {
    var _addEventListener;....
```

Such a site is report-uri.io.

Vimium works fine nonetheless.

**Comment by gdh1995** [comment]
Yes, the injected part can be rejceted on Chrome, too, though it matters little about the main functionality of `LinkHints.*` commands.

As far as I know, it seems no method to detect such CSP info to avoid a real injection, so it might be always there, and we had better ignore it.

**Comment by lydell** [comment]
There might be an alternative way in Firefox: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Sharing_objects_with_page_scripts
